LBHackney-IT/lbhackney-it.github.io
# Migrate content from sites that form playbook

This PR migrates content from the following repositories:

- https://LBHackney-IT/API-Playbook
- https://LBHackney-IT/api-specifications
- https://LBHackney-IT/architecture-pillars
- https://LBHackney-IT/micro-frontends
- https://LBHackney-IT/Product-Playbook
- https://LBHackney-IT/ways-of-working

To be considered in conjunction with #14, which details the rationale for the migration and sets up the pre-requisites.

## Migrating content

Content has been copied over as-is where possible. No effort has been made to fix existing formatting or structure quirks.

A handful of fixes were made:

1. Switch some indented code blocks for fenced code blocks. Some indented code blocks broke the MDX parser. Swapping these to fenced code blocks fixed the error and allowed us to add the content type, which improves syntax highlighting.
2. Removed `slug` frontmatter, because it was causing pages to redirect incorrectly. Slugs aren't necessary when the file name matches the intended path.
3. Removed redundant frontmatter. IDs and titles were removed in preference for standard markdown headings.
4. Image links were fixed where the move caused a problem

### URL structure

In general the structure has been moved across as-is. However, several of the sites used a more complex approach to sidebar and URL management. This has been stripped out to favour simplicity, so the entire site now relies on auto-generated sidebars based on the directory structure. This means some permalinks will be broken by this migration, as they are updated to reflect their position in the directory structure.

## Deleting content

Although some content appears out of date, the only content that's been removed is the duplicated "contact us" pages present on most of the repositories. If you notice any obviously dated content during your review please highlight it and we can remove it in a subsequent PR.

## Product Playbook

The Product Playbook has been migrated as part of this PR, however Product and Engineering are separate disciplines so this _may_ warrant its own site on a separate domain.

Conversely, it's only engineers that make changes to the Product Playbook at the moment so consolidating into one repo would reduce the upgrade burden.

Alternative options are:

1. Move the Product Playbook to a new domain (product-playbook.hackney.gov.uk for example) and continue using a separate repository. This fits the organisation structure, but adds another repository to update dependencies on.
2. Leave the Product Playbook in https://github.com/LBHackney-IT/Product-Playbook as a section of playbook.hackney.gov.uk. This is the least-preferred option as it mixes architectures and the header/navigation will become inconsistent. Links from the main engineering site to the product playbook (if any exist) may break because the Docusaurus SPA has its own router. 

## Data Platform Playbook

The Data Platform Playbook isn't included here (yet). The same options exist as for the Product Playbook, though the Data Team are able to update and maintain the site themselves. Therefore keeping the site separate and creating a new domain for it might be the best route. 

I've asked the team what route they'd prefer.

## How to test this PR

There's a lot of content here, so it's not feasible to check every single line. Instead I'd suggest:

1. Check out the branch and run `npm install`, then `npm run build`, then `npm run serve`. This checks the build process and generated content (rather than the dev live-reloading site).
2. Check a random sample from each section of the site (in the header). Compare the sidebar structure with playbook.hackney.gov.uk, and check that the page loads.
3. Check a random sample of internal links.
4. Check the nav bar links behave as expected.

## How to merge this PR

- See #14 for merge steps
- Once this PR is live, the repositories listed above should be marked as end-of-life, removed from GitHub Pages, and archived.

## Sample screenshots
![Screenshot 2024-10-17 at 11 48 (Brave Browser)@2x](https://github.com/user-attachments/assets/eac45442-15ed-40c1-9232-c2a97f5308ba)

![Screenshot 2024-10-17 at 11 49 (Brave Browser)@2x](https://github.com/user-attachments/assets/dc3e5df9-fdbb-4271-825b-11671f022838)

![Screenshot 2024-10-17 at 11 49 (Brave Browser) 2@2x](https://github.com/user-attachments/assets/08b750af-9e99-4376-91df-9122624ff2aa)
